File: 345.py
# working code <ode
def reverseVowels(s):
    # converting into list 
    # string is immutable
    s=list(s)
    charVowels = 'aeiouAEIOU'
    left = 0
    right = len(s)-1
    while(left<right):
        while(left<right and s[left] in charVowels):
            left+=1
        while(left<right and s[right] in charVowels):
            right-=1
        if(left<right):
            s[left],s[right]=s[right],s[left]
            left+=1
            right-=1
    return ''.join(s) 

File: 345.cpp
#include<bits/stdc++.h>
using namespace std;

bool isVowel(char c){
  return (c=='a' || c=='A' || c=='E' || c=='e' || c=='I' || c=='i' || c=='O' || c=='o' || c=='u' || c=='U');
}

string reverseVowels(string s){
  int left =0;
  int right = s.size()-1;
  while(left < right){
    while(left<right  && !isVowel(s[left])){
      left++;
    }
    while(left<right && !isVowel(s[right])){
      right--;
    }
    if(left<right){
      swap(s[left],s[right]);
        left++;
        right--;
    }
  }
  return s;
}

int main()
{
  string s="hello";
  string res = reverseVowels(s);
  cout<<res;
  return 0;
}
